src/main/js/html.js

```diff
@@ -921,35 +928,44 @@ var backgroundColorAdjustSuffix = "BackgroundColorAdjust";
 
                     if (s * (e.after - frontier) < 0) {
 
-                        pad = Math.ceil(Math.abs(frontier - e.after)) + "px";
-
-                        e.node.style.backgroundColor = e.bgcolor;
+                        pad = Math.abs(frontier - e.after);
 
+                        if (pad>maxPad) {
+                            maxPad=pad;
+                        }
+                    }
+                }
+                thisNode=element.getElementsByTagName("span")[i];
+                if (!lineList[i] || thisNode.childElementCount==lineList[i].elements.length) {
+// this works for m000sm34
```

Comment:
What behaviour was observable with that pid before that this code change fixes?

Reply:
There's two different results in our renderings depending on how the span children are laid out, one results in an extra span around all the children, and that is the one which should have the line-padding adjustments to, or the line padding should go on the next level down on an individual character basis.  The decision of which is to use depends on noticing the layout - which the check does, without it you get overlaps or gaps depending on which strategy you use, it's to do with how the background colour is applied.

I've no guarantee that the detection is robust in other options, and there's a bug with the linegap test where the calculation for the size of the padding is way too high (but in the right places if it was smaller) which I've not investigated yet, but it does a good job with our subs.